### ISSUE TITLE
KGL Nehe Examples (16 + 26 at least) Not Working

### ISSUE THREAD

**Issue by gyrovorbis** [KallistiOS/KallistiOS#445]
While working on https://github.com/KallistiOS/KallistiOS/pull/444, I realized that some of the Nehe examples weren't working properly... On *at least* Nehe 16 and 26 for KGL, I lose video output as soon as sending the ELF, and I *believe* it stops responding, since I cannot quit the application... Not sure when this happened or if it impacts all of the Nehe examples for KGL. Can someone double check and/or diagnose this?

NOTE: Equivalent examples for GLdc are working fine.

**Comment by andressbarajas** [comment]
All KGL examples were broken for me.  Here is the fix: https://github.com/KallistiOS/libkgl/pull/1

**Comment by gyrovorbis** [comment]
I just posted a novel on what the deal is in Discord, but I'll echo the main points here for everyone... There are two issues now that have broken all of KGL and its examples:

1) When this commit (rightfully) migrated the SQ usage away from being direct and towards the new API, it looks like it is accidentally passing the SQ masked and shifted destination address as the destination input to `sq_copy()`, which is itself then trying to recompute the SQ_DEST address again. 

https://github.com/KallistiOS/libkgl/commit/4c8c082e0675162a1482a810210dec6d4202dc3e

The solution is trivial and is just to use the raw `PVR_TA_INPUT` register address as the destination to `sq_copy()` or maybe better yet, `pvr_sq_cpy()`. 

2) The REAL problem... even after fixing that, everything is still 100% broken and will present  you with a nice black screen. The underlying issue appears to be in the fast-path of `sq_copy()` which is utilizing Moop's SH4 ASM optimized double-FP SQ transfers... 

When KGL is submitting its vertices, they're all 32-byte aligned, so it's always using that fast path, which is what breaks. I can literally remove the fast path and only use the default old C path, and it fixes everything. I've tried multiple times to implement the C equivalent of the fast ASM code from Moop with the resulting ASM being HORRIBLE even if it all worked fine... but I cannot for the life of me figure out what the fuck is wrong with his logic... 

This is where it is in the codebase: https://github.com/KallistiOS/KallistiOS/blob/1162af57fab083584c7b6c0e553cbcb15723424a/kernel/arch/dreamcast/hardware/sq.c#L91
